### ISSUE TITLE
Adding more OAuth scopes when needed?

### ISSUE THREAD

**Issue by mikabu** [playgameservices/play-games-plugin-for-unity#1658]
Is there a way to request for additional permissions when the user accesses certain features of the app (when Activate() has been called once already)?

PlayGamesClientConfiguration.Builder().AddOauthScope(scope)
- I believe you can only do this before calling Activate() and you can only call Activate() once during the app's lifetime

Do you think it would be hard to add this functionality?
Thanks.

**Comment by claywilkinson** [comment]
Yes, it is hard.  The client connection is created just once per instance.  To do it dynamically you would need to reload the app.

**Comment by claywilkinson** [comment]
I am also curious why you want to not add all the scopes at the beginning?  This seems like a better user experience since they would consent to all scopes the first time they launch, and then never again?

**Comment by mikabu** [comment]
@claywilkinson thanks for the clarification.
In our case it's not immediately obvious to the user why we need the particular access since the feature needing it, is entirely optional and many users might not be even using the service.

Also, the recommended way is to do it incrementally:
“It is generally a best practice to request scopes incrementally, at the time access is required, rather than up front.”
https://developers.google.com/identity/protocols/OAuth2
